hashing/src/chunk.rs

```diff
@@ -24,6 +59,20 @@ impl<const N: usize> ChunkWithProof<N> {
             .to_vec();
         Ok(ChunkWithProof { proof, chunk })
     }
+
+    pub fn is_valid(&self) -> bool {
+        let chunk_hash = blake2b_hash(self.chunk());
+        self.proof
+            .merkle_proof()
+            .iter()
+            .next()
+            .map_or(false, |first_hash| chunk_hash == *first_hash)
+    }
+
+    /// Get a reference to the chunk with proof's chunk.
+    pub fn chunk(&self) -> &[u8] {
+        self.chunk.as_slice()
+    }
 }
 
```

Comment:
```suggestion
const CHUNK_SIZE: usize = 1_048_576;   // 1 MiB

type CheckWithProof = SizedChunkWithProof<CHUNK_SIZE>;
```

Reply:
Reverted to the initial solution with uses the `#[cfg(test)]` attribute.